### PR TITLE
[DNM] Resharper cleanup

### PR DESCRIPTION
**Pull request by nschonni**
Through the base through ReSharper to see what it could come up with. There is some other stuff, but this was the simplest pieces to see if there is an interest.

### CODE REVIEW

**Comment by mgreter** [comment]
Looks fine for me. I actually expected some warnings in clang or gcc, but they all seem to agree. Not sure if some older compilers will warn missing return when a throw is in place. But CI seems to agree here!

**Comment by nschonni** [comment]
@mgreter I left all the commits discrete if you want to cherrypick parts, or I can fastforward the branch with the latest.
PS: There were more warnings, these where just the simple stuff

**Comment by xzyfer** [comment]
There are bits and pieces I'll cherry-pick onto to master. I'll take care
of it this weekend. Thanks Nick.

Regards,
Michael

**Comment by xzyfer** [comment]
@nschonni there has been a lot of churn on master. Want to give this another shot?

**Comment by nschonni** [comment]
@xzyfer you mentioned you wanted to cherry-pick some stuff. I can do specific PRs for the parts if you call them out here

**Comment by am11** [comment]
Nice cleanup! 👍

Related to this, would it be a good idea to have [`.clang-format`](http://clang.llvm.org/docs/ClangFormat.html) rules file like [this](https://github.com/dotnet/coreclr/blob/36a2b90/src/jit/.clang-format) and make TravisCI do the linting based on the clang-format per PR like [these guys](https://github.com/citra-emu/citra/blob/6e5e5be/.travis-build.sh) are doing? Then in our contribution guide, we can say something this effect:

> To avoid build failures due to code formatting, make sure to run `.clang-format` before issuing the pull request. On Windows, you can use [Clang-Format](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format) extension for VS.

**Comment by nschonni** [comment]
Rebased, but I'm going to need to go back and redo some stuff because of the switch to using `Cast`

**Comment by xzyfer** [comment]
@am11 omg I've been dreaming about something like `.clang-format`. I couldn't find docs on a prevailing convention or tool for this stuff.

File: src/to_value.cpp [hunk 1]
```diff
@@ -9,8 +9,6 @@ namespace Sass {
     // throw a runtime error if this happens
     // we want a well defined set of possible nodes
     throw std::runtime_error("invalid node for to_value");
-    // mute warning
```
Comment (mgreter) [inline src/to_value.cpp]: Interestingly there seems to have been a reason for these superflous returns ...

**Comment by mgreter** [comment]
Rebased and cherry picked in https://github.com/sass/libsass/pull/2316

**Comment by mgreter** [comment]
Closing this as most has been merged with https://github.com/sass/libsass/pull/2316. Thanks @nschonni!

**Pull request closed without merging**